### ISSUE TITLE
Change to Ruby URI gem breaks unix:// URIs

### ISSUE THREAD

**Issue by jeffparsons** [excon/excon#805]
This change makes the URI `host` field an empty string instead of `nil` when absent: https://github.com/ruby/uri/commit/81263c9e94bd67ca01deee238842a88c2c8885f3

This line in excon appears to assume it will be nil: https://github.com/excon/excon/blob/efd48747fe6c6fa959e787aa5949241cd762f8f3/lib/excon/connection.rb#L92

Discovered via 'docker-api' gem. Example IRB session:

```
irb(main):001:0> require 'docker-api'
=> true
irb(main):002:0> Docker.url = 'unix:///var/run/docker.sock'
=> "unix:///var/run/docker.sock"
irb(main):003:0> Docker::Container::all
/home/jeff/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/excon-0.95.0/lib/excon/connection.rb:93:in `initialize': The `:host` parameter should not be set for `unix://` connections. (ArgumentError)                     
When supplying a `unix://` URI, it should start with `unix:/` or `unix:///`.                    
        from /home/jeff/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/excon-0.95.0/lib/excon.rb:142:in `new'
        from /home/jeff/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/excon-0.95.0/lib/excon.rb:142:in `new'
        from /home/jeff/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/docker-api-2.2.0/lib/docker/connection.rb:35:in `resource'                                                                                         
        from /home/jeff/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/docker-api-2.2.0/lib/docker/connection.rb:45:in `request'                                                                                          
        from /home/jeff/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/docker-api-2.2.0/lib/docker/connection.rb:109:in `block (2 levels) in <class:Connection>'                                                          
        from /home/jeff/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/docker-api-2.2.0/lib/docker/container.rb:343:in `all'                                                                  
        from (irb):3:in `<main>'                                    
        from /home/jeff/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/irb-1.6.2/exe/irb:11:in `<top (required)>'
	from /home/jeff/.rbenv/versions/3.2.0/bin/irb:25:in `load'
	from /home/jeff/.rbenv/versions/3.2.0/bin/irb:25:in `<main>'
```

**Comment by jeffparsons** [comment]
This is in uri version 0.12.0: https://github.com/ruby/uri/releases/tag/v0.12.0

This version is a default gem in Ruby 3.2 (which was released a few days ago).

**Comment by jeffparsons** [comment]
I realised that docker-api example might not be all that useful. I will put together a minimal excon-only reproduction.

**Comment by jeffparsons** [comment]
This is equivalent to what docker-api asks excon to do, but as a standalone example:

```
irb(main):002:0> Excon.new('unix:///', {socket: '/tmp/foo'})
/home/jeff/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/excon-0.95.0/lib/excon/connection.rb:93:in `initialize': The `:host` parameter should not be set for `unix://` connections. (ArgumentError)
When supplying a `unix://` URI, it should start with `unix:/` or `unix:///`.
        from /home/jeff/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/excon-0.95.0/lib/excon.rb:142:in `new'
        from /home/jeff/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/excon-0.95.0/lib/excon.rb:142:in `new'
        from (irb):2:in `<main>'                                         
        from /home/jeff/.rbenv/versions/3.2.0/lib/ruby/gems/3.2.0/gems/irb-1.6.2/exe/irb:11:in `<top (required)>'
        from /home/jeff/.rbenv/versions/3.2.0/bin/irb:25:in `load'       
        from /home/jeff/.rbenv/versions/3.2.0/bin/irb:25:in `<main>'     

```

**Comment by geemus** [comment]
Fixed by #807 and released in v.096.0
